### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by phillbaker**
This handles non 200-299 responses by all resources with specific errors, where necessary and default handlers.

Addresses #59 and the rest of #1.

cc @nanzhong

### CODE REVIEW

**Comment by route** [comment]
Any updates on this? It was confusing when response was `"{\"id\":\"unauthorized\",\"message\":\"Unable to authenticate you.\"}"` but it showed `TypeError: no implicit conversion of String into Array`

File: lib/droplet_kit/paginated_resource.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ def retrieve(page, per_page = self.per_page)
       invoker = ResourceKit::ActionInvoker.new(action, resource, *@args)
       invoker.options[:per_page] ||= per_page
       invoker.options[:page]       = page
-
+      # require 'pry'; binding.pry
```
Comment (nanzhong) [inline lib/droplet_kit/paginated_resource.rb]: Extra debug comment leftover?

File: lib/droplet_kit.rb [hunk 2]
```diff
@@ -62,10 +62,27 @@ module DropletKit
...
+
+    def limit
+      @limit.to_i if @limit
+    end
+
+    def remaining
+      @remaining.to_i if @remaining
+    end
+
+    def reset_at
```
Comment (nanzhong) [inline lib/droplet_kit.rb]: Would it make sense to keep the symmetry of the naming consistent here?

**Comment by nanzhong** [comment]
:+1:, just a couple of small comments.

File: lib/droplet_kit.rb (hunk 2 above)
Comment (phillbaker) [inline lib/droplet_kit.rb]: Whoops, fixed - I had some vague memory of where the client converted timestamp strings to objects, but I couldn't find any after looking, so removed this.

**Comment by nanzhong** [comment]
:ship:

**Pull request merged**